tree/treeplayer/src/TRootDS.cxx

```diff
@@ -13,8 +13,19 @@ namespace ROOT {
 namespace Experimental {
 namespace TDF {
 
-std::vector<void *> TRootDS::GetColumnReadersImpl(std::string_view name, const std::type_info &)
+std::vector<void *> TRootDS::GetColumnReadersImpl(std::string_view name, const std::type_info &id)
 {
+   const auto colTypeName = GetTypeName(name);
+   const auto &colTypeId = ROOT::Internal::TDF::TypeName2TypeID(colTypeName);
+   if (id != colTypeId) {
+      std::string err = "The type of column \"";
+      err += name;
+      err += "\" is ";
+      err += colTypeName;
+      err += " but a different one has been selected.";
```

Comment:
maybe `id.name()` would give users an extra hint of what they did wrong -- or maybe it would confuse them more 😅